layouts/hub-page.tsx

```diff
@@ -9,9 +9,11 @@ import InsightHeader from "components/molecules/InsightHeader/insight-header";
 
 import useNav from "lib/hooks/useNav";
 import useInsight from "lib/hooks/useInsight";
+import useSupabaseAuth from "lib/hooks/useSupabaseAuth";
 
 const HubPageLayout = ({children}: {children: React.ReactNode}) => {
   const router = useRouter();
+  const { user } = useSupabaseAuth();
   const { filterName } = router.query;
   const insightId = filterName as string;
   const { data: insight } = useInsight(insightId);
```

Comment:
This is not checking for errors; checking for errors, the `insight?.user_id` can be simplified to `insight.user_id`

Reply:
How should I check for errors? You mean like put the logic separately in a variable and pass it instead?
```ts
let isOwner = insight && (userId === insight.user_id)
```
?